src/exporter/DaemonMetricCollector.cc

```diff
@@ -80,7 +81,23 @@ std::string boost_string_to_std(boost::json::string js) {
 
 std::string quote(std::string value) { return "\"" + value + "\""; }
 
-bool is_hyphen(char ch) { return ch == '-'; }
+void promethize(std::string &name) {
```

Comment:
I'll approve whatever this function does if you unit-test it by taking a dump of all (current) perfcounters (or a neatly chosen subset of those) and this outputs the exact same metric names as the [`mgr/prometheus` exporter](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels).

It should be mandatory to write unit tests for pure functions.

Reply:
I've decided to copy all the metrics I could gather from a `vstart.sh` cluster `MON=1 OSD=1 MDS=1 FS=1 NFS=1 RGW=1` as parameters. With  ~600 metrics I tested the function based on how the `prometheus/module.py` performs it. I hope this helps.

Reply:
👍🏼